File: android/app/src/main/java/com/rcm/dealer/NativeTTSHelper.java
package com.rcm.dealer;

import android.content.Context;
import android.speech.tts.TextToSpeech;
import android.webkit.JavascriptInterface;

import java.util.Locale;

public class NativeTTSHelper {

    private TextToSpeech tts;
    private Context context;

    NativeTTSHelper(Context context) {
        this.context = context;
        tts = new TextToSpeech(context, status -> {
            if (status == TextToSpeech.SUCCESS) {
                tts.setLanguage(Locale.getDefault());
            }
        });
    }

    @JavascriptInterface
    public void speak(String text) {
        tts.speak(text, TextToSpeech.QUEUE_FLUSH, null, null);
    }
}

File: android/app/src/main/java/com/rcm/dealer/NativeTTSHelper.kt

package com.rcm.dealer

import android.content.Context
import android.speech.tts.TextToSpeech
import android.webkit.JavascriptInterface
import java.util.Locale

class NativeTTSHelper(context: Context) : TextToSpeech.OnInitListener {

    private var tts: TextToSpeech = TextToSpeech(context, this)
    private var isReady: Boolean = false

    @JavascriptInterface
    fun speak(text: String) {
        if (isReady) {
            tts.speak(text, TextToSpeech.QUEUE_FLUSH, null, "")
        }
    }

    override fun onInit(status: Int) {
        if (status == TextToSpeech.SUCCESS) {
            tts.language = Locale("hi", "IN")
            tts.setPitch(1.4f) // Young female pitch
            isReady = true
        }
    }
}
